backend/routes/mcp_servers.py

```diff
@@ -1,118 +1,257 @@
+"""
+Model Context Protocol (MCP) Server management API endpoints.
+
+This module provides CRUD operations for MCP servers, which are external tools
+and services that can be integrated with AI agents through the Model Context Protocol.
+MCP servers are automatically discovered and synchronized with LlamaStack's tool system.
+
+Key Features:
+- Register and manage MCP server configurations
+- Automatic synchronization with LlamaStack tool discovery
+- Tool group management and parameter configuration
+- Integration with virtual assistants for enhanced capabilities
+"""
+
+from typing import List
+
 from fastapi import APIRouter, Depends, HTTPException, status
 from sqlalchemy.ext.asyncio import AsyncSession
 from sqlalchemy.future import select
-from typing import List, Dict, Any
 
 from .. import models, schemas
-from ..database import get_db
 from ..api.llamastack import client
+from ..database import get_db
+from ..utils.logging_config import get_logger
+
+logger = get_logger(__name__)
 
 router = APIRouter(prefix="/mcp_servers", tags=["mcp_servers"])
 
-@router.post("/", response_model=schemas.MCPServerRead, status_code=status.HTTP_201_CREATED)
-async def create_mcp_server(server: schemas.MCPServerCreate, db: AsyncSession = Depends(get_db)):
+
+@router.post(
+    "/", response_model=schemas.MCPServerRead, status_code=status.HTTP_201_CREATED
+)
+async def create_mcp_server(
+    server: schemas.MCPServerCreate, db: AsyncSession = Depends(get_db)
+):
+    """
+    Create a new MCP server configuration.
+
+    This endpoint registers a new MCP server in the database and triggers
```

Comment:
I believe mcp servers need to be registered with llama stack in order for tool discovery to happen. Is there a reason for maintaining a list of mcp servers in the database when we can get them from llama stack and forgo the need to keep them in “sync” ?